uint16[4] temp_velocity
int32[4] temp_position

